tasks/push-snapshot/push-snapshot.yaml

```diff
@@ -122,15 +135,25 @@ spec:
               fi
             fi
             if [[ $(jq -r ".images.addSourceShaTag" "${DATA_FILE}") != "false" ]] ; then # Default to true
-              if [[ "${containerImage}" == *"@sha256"* && $(echo "${containerImage}" | tr -cd ':' | wc -c) -eq 1 ]]
-              then
-                sha=$(echo "${containerImage}" | cut -d ':' -f 2)
+              if [ -n "$sha" ]; then
                 push_image "${name}" "${containerImage}" "${repository}" "${sha}"
               else
                 printf 'Asked to create source sha based tag, but no sha found in %s\n' "${containerImage}"
                 exit 1
               fi
             fi
+            # Push the associated source container using the common tag
+            if [[ "$(params.pushSourceContainer)" == "true" ]] ; then # Default to false
+              # Calculate the source container image based on the provided container image
+              sourceContainer="${containerImage%@sha256:*}@sha256:${sha}.src"
+              # Check if the source container exists
+              if ! skopeo inspect "docker://${sourceContainer}" &>/dev/null; then
```

Comment:
@johnbieren the Tekton test is failing at this step, and if I comment on this call, it passes. 
I cannot figure out what exact changes I need to make in `mocks.sh` for passing this test.

logs from local run
```
[run-task : push-snapshot] + [[ true == \t\r\u\e ]]
[run-task : push-snapshot] + sourceContainer=registry.io/image@sha256:abcdefg.src
[run-task : push-snapshot] + skopeo inspect docker://registry.io/image@sha256:abcdefg.src
[run-task : push-snapshot] + echo Mock skopeo called with: inspect docker://registry.io/image@sha256:abcdefg.src
[run-task : push-snapshot] + echo inspect docker://registry.io/image@sha256:abcdefg.src
[run-task : push-snapshot] Mock skopeo called with: inspect docker://registry.io/image@sha256:abcdefg.src
[run-task : push-snapshot] + [[ inspect docker://registry.io/image@sha256:abcdefg.src != \i\n\s\p\e\c\t\ \-\-\n\o\-\t\a\g\s\ \-\-\f\o\r\m\a\t\ \{\{\.\D\i\g\e\s\t\}\}\ \d\o\c\k\e\r\:\/\/* ]]
[run-task : push-snapshot] + echo Error: Unexpected call
[run-task : push-snapshot] Error: Unexpected call
[run-task : push-snapshot] + exit 1
```

Reply:
NVM, I updated the `mocks.sh` to support new `skopeo` call.